esphome/components/camera/__init__.py

```diff
@@ -1,10 +1,63 @@
+from esphome import automation
 import esphome.codegen as cg
 import esphome.config_validation as cv
+from esphome.const import CONF_TRIGGER_ID
 
 CODEOWNERS = ["@DT-art1"]
 
-CONFIG_SCHEMA = cv.Schema({})
+CONF_ON_STREAM_START = "on_stream_start"
+CONF_ON_STREAM_STOP = "on_stream_stop"
+CONF_ON_IMAGE = "on_image"
+
+camera_ns = cg.esphome_ns.namespace("camera")
+Camera = camera_ns.class_("Camera", cg.Component, cg.EntityBase)
+CameraImageData = camera_ns.struct("CameraImageData")
+CameraImageTrigger = camera_ns.class_(
+    "CameraImageTrigger", automation.Trigger.template()
+)
+CameraStreamStartTrigger = camera_ns.class_(
+    "CameraStreamStartTrigger",
+    automation.Trigger.template(),
+)
+CameraStreamStopTrigger = camera_ns.class_(
+    "CameraStreamStopTrigger",
+    automation.Trigger.template(),
+)
+
+CONFIG_SCHEMA = cv.Schema(
```

Comment:
This should not be named `CONFIG_SCHEMA` as that indicates this is a component that has its own configuration in yaml
```suggestion
CAMERA_SCHEMA = cv.Schema(
```